websites/G/G2G/presence.ts

```diff
@@ -0,0 +1,97 @@
+const presence = new Presence({
+		clientId: "1073167802296438875",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "https://i.imgur.com/XkzTOsp.png",
+		startTimestamp: browsingTimestamp,
+	};
+
+	switch (document.location.pathname.split("/")[1]) {
+		case "search": {
+			presenceData.details = "Searching something to buy";
+			presenceData.state = new URLSearchParams(document.location.search).get(
+				"q"
+			);
+			break;
+		}
+		case "categories": {
+			presenceData.details = "Browsing a category";
+			presenceData.state =
+				document.querySelector<HTMLHeadingElement>("title").textContent;
+			presenceData.buttons = [
+				{
+					label: "View Category",
+					url: document.location.href,
+				},
+			];
+			break;
+		}
+		case "offer": {
+			presenceData.details = "Viewing an offer";
+			presenceData.state =
+				document.querySelector<HTMLHeadingElement>("title").textContent;
+			presenceData.buttons = [
+				{
+					label: "View Offer",
+					url: document.location.href,
+				},
+			];
+			break;
+		}
+		case "seller": {
+			presenceData.details = "Want to sell something";
+			presenceData.state = "Browsing...";
+			break;
+		}
+		case "chat": {
+			presenceData.details = "Chatting with a seller";
+			presenceData.state = "Browsing...";
+			break;
+		}
+		case "account": {
+			presenceData.details = "Viewing their account";
+			presenceData.state = "Browsing...";
+			break;
+		}
+		case "trending": {
+			presenceData.details = "Viewing a trending offer";
+			presenceData.state =
+				document.querySelector<HTMLBodyElement>("div.text-h4").textContent;
```

Comment:
The wrong type is used in this query selector (and other ones)
```suggestion
				document.querySelector<HTMLDivElement>("div.text-h4").textContent;
```